app/components/widgets/AlertBox.vue.ts

```diff
@@ -0,0 +1,166 @@
+  @Inject() alertBoxService: AlertBoxService;
+
+  $refs: { [key: string]: HTMLElement };
+
+  afterFetch() {
+    this.alertTypes = this.alertTypes.filter((type) => this.wData.settings[type]);
+  }
+
+  alertName(alertType: string) {
+    return alertNameMap()[alertType];
+  }
+
+  alertTypes = this.alertBoxService.apiNames();
+
+  addAlertMenuOpen = false;
+  selectedAlert = 'general';
+  selectedId = 'default';
+  editingName: string = null;
+
+  get selectedVariation() {
+    if (this.selectedAlert === 'general') { return this.wData }
+    return this.wData.settings[this.selectedAlert].variations.find(
+      (variation: IAlertBoxVariation) => variation.id === this.selectedId
+    );
+  }
+
+  get navItems() {
+    if (this.selectedAlert === 'general') {
+      return [
+        { value: 'general', label: $t('General Settings') },
+        { value: 'moderation', label: $t('Moderator Tools') },
+        { value: 'source', label: $t('Source') }
+      ];
+    }
+    const baseItems = [
+      { value: 'title', label: $t('Title Message') },
+      { value: 'media', label: $t('Media') },
+      { value: 'animation', label: $t('Animation') }
+    ];
+    if (this.selectedVariation.settings.message) {
+      baseItems.push({ value: 'message', label: $t('Donor Message') })
+    }
+    if (['donations', 'bits', 'hosts', 'raids'].includes(this.selectedAlert) || this.selectedId !== 'default') {
+      baseItems.push({ value: 'alert', label: $t('Alert Settings') })
+    }
+    return baseItems;
+  }
+
+  get conditions() {
+    return this.alertBoxService.conditionsByType(this.selectedAlert);
+  }
+
+  get minTriggerAmount() {
+    return this.wData.settings[triggerAmountMap[this.selectedAlert]];
+  }
+
+  set minTriggerAmount(value: number) {
+    this.wData.settings[triggerAmountMap[this.selectedAlert]] = value;
+  }
+
+
+  get minRecentEvents() {
+    return this.selectedAlert === 'donation' ?
+      this.wData.settings.recent_events_donation_min_amount : this.wData.settings.recent_events_host_min_viewer_count;
+  }
+
+  set minRecentEvents(value: number) {
+    if (this.selectedAlert === 'donation') {
+      this.wData.settings.recent_events_donation_min_amount = value;
+    } else {
+      this.wData.settings.recent_events_host_min_viewer_count = value;
+    }
+  }
+
+  selectAlertType(alertName: string) {
+    this.selectedId = 'default';
+    this.selectedAlert = this.selectedAlert === alertName ? 'general' : alertName;
+  }
+
+  selectVariation(id: string) {
+    this.selectedId = id;
+  }
+
+  toggleAddAlertMenu() {
+    this.addAlertMenuOpen = !this.addAlertMenuOpen;
+  }
+
+  addAlert(type: string) {
+    const newVariation = this.alertBoxService.newVariation(type);
+    this.wData.settings[type].variations.push(newVariation);
+    this.selectedAlert = type;
+    this.addAlertMenuOpen = false;
+    this.save();
+    this.$nextTick(() => this.editName(newVariation.id));
```

Comment:
why we need nextTick here? `editName()` already has it

Reply:
we push in a new variation here and have to wait till it renders on the dom before we can select said variation

Reply:
@holiber thanks for pointing out that UX issue
I am going to think of a solution